### PR TITLE
Three column layout (rebased/squashed)

### PR DESCRIPTION
**Pull request by juliushaertl**
Rebased and squashed version of #842 to have an easier way to resolve the conflicts

So far pushed another commit additionally to bring back the actions menu in the list after my rebase and some minor styling fixes

Screenshot 2023-04-29 at 12 28 00

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -61,8 +58,8 @@ import PlusIcon from 'vue-material-design-icons/Plus.vue'
 import CogIcon from 'vue-material-design-icons/Cog.vue'
 
 import AppSettings from './components/AppSettings.vue'
-import NavigationList from './components/NavigationList.vue'
 import EditorHint from './components/Modal/EditorHint.vue'
+import CategoriesList from './components/CategoriesList.vue'
```
Comment (JonnyTischbein) [inline src/App.vue:62]: I would move it to line 61 to keep the ordering

File: src/App.vue [hunk 2]
```diff
@@ -74,14 +71,14 @@ export default {
...
 		EditorHint,
-		NavigationList,
 		NcAppContent,
 		NcAppNavigation,
 		NcAppNavigationNew,
 		NcAppNavigationItem,
 		NcContent,
 		CogIcon,
 		PlusIcon,
+		CategoriesList,
```
Comment (JonnyTischbein) [inline src/App.vue:81]: Again order it alphabetically like in the other files

File: src/components/CategoriesList.vue [hunk 3]
```diff
@@ -1,15 +1,8 @@
...
-		:class="{ 'category-header': selectedCategory !== null }"
-		:open.sync="open"
-		:allow-collapse="true"
-		@click.prevent.stop="onToggleCategories"
-	>
-		<FolderIcon slot="icon" :size="20" />
+	<Fragment>
 		<NcAppNavigationItem
 			:title="t('notes', 'All notes')"
+			:class="{ active: null === selectedCategory }"
```
Comment (JonnyTischbein) [inline src/components/CategoriesList.vue:5]: Shouldn't it be `selectedCategory === null` ?

File: src/components/NoteItem.vue [hunk 4]
```diff
@@ -27,29 +41,29 @@
...
 export default {
-	name: 'NavigationNoteItem',
+	name: 'NoteItem',
 
 	components: {
 		NcActionButton,
-		NcActionSeparator,
-		NcAppNavigationItem,
+		NcListItem,
+		AlertOctagonIcon,
```
Comment (JonnyTischbein) [inline src/components/NoteItem.vue:64]: I would fix the order again

File: src/components/NoteItem.vue [hunk 5]
```diff
@@ -64,50 +78,46 @@ export default {
...
 		},
-
 		actionDeleteIcon() {
 			return 'icon-delete' + (this.loading.delete ? ' loading' : '')
 		},
 	},
 
 	methods: {
+		onNoteSelected(noteId) {
+			console.error('select note', noteId)
```
Comment (JonnyTischbein) [inline src/components/NoteItem.vue:118]: Is this a debug log ?

File: src/components/NotesView.vue [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+		category() {
+			return store.getters.getSelectedCategory()
+		},
+
+		filteredNotes() {
+			return store.getters.getFilteredNotes()
+		},
+
+		displayedNotes() {
+			if (this.filteredNotes.length > 40 && this.showFirstNotesOnly) {
```
Comment (JonnyTischbein) [inline src/components/NotesView.vue:113]: Why you catch for `> 40` but slice to `0-30` ?

File: src/components/NotesView.vue [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+
+		categoryToLabel(category) {
+			return categoryLabel(category.substring(this.category.length + 1))
+		},
+
+		getTimeslotFromNote(note) {
+			if (note.favorite) {
+				return ''
+			}
+			const t = note.modified * 1000
```
Comment (JonnyTischbein) [inline src/components/NotesView.vue:186]: May move `1000` to const variable

File: src/components/Sidebar.vue [hunk 8]
```diff
@@ -170,6 +195,22 @@ export default {
 				})
 		},
 
+		onUpdateTitle(newTitle) {
+			this.newTitle = newTitle
+		},
+
+		onRenameTitle(event) {
+			if (event && this.title !== this.newTitle) {
```
Comment (JonnyTischbein) [inline src/components/Sidebar.vue:203]: Is  it possible that an event is triggered, that is not defined ?
I think it is safe to remove `event` from the if clause

**Review by JonnyTischbein — COMMENTED** [review]
I will test the function tomorrow, but here some comments on the code

File: src/components/NotesView.vue (hunk 7 above)
Comment (juliushaertl) [reply src/components/NotesView.vue:186]: I'd still keep this as this is a pattern that js developers likely know well and a constant (or timestampToMilliseconds helper that I was thinking about) would be easy to be overseen in new code introduced and then we'd have two ways of doing that.

File: src/components/NotesView.vue (hunk 6 above)
Comment (juliushaertl) [reply src/components/NotesView.vue:113]: Something to consider separately I'd say as it is not introduced here but originating from https://github.com/nextcloud/notes/pull/424/files#diff-84c733529b3b3fd6edfc369f85fe95b5cfb8d1f0a74d39712293aa3b037a0848R113-R117 s

File: src/components/CategoriesList.vue (hunk 3 above)
Comment (juliushaertl) [reply src/components/CategoriesList.vue:5]: Yep, more what the current code base and Nextcloud apps are using.

**Comment by juliushaertl** [comment]
Thanks for the review @JonnyTischbein, very much appreciated. I addressed most of them, two I'd keep out of this PR.

File: src/components/Note.vue [hunk 9]
```diff
@@ -1,6 +1,6 @@
 <template>
-	<NoteRich v-if="rich" :note-id="noteId" />
-	<NotePlain v-else :note-id="noteId" />
+	<NoteRich v-if="ready && rich" :note-id="noteId" />
```
Comment (luka-nextcloud) [inline src/components/Note.vue:2]: I think `v-if="rich"` is enough here because rich already includes condition check of ready (check line 27 and 30).

Comment (juliushaertl) [reply src/components/Note.vue]: Thanks, pushed a bit of a change there to make the state more clear through computed properties

**Review by luka-nextcloud — APPROVED** [review]

**Comment by juliushaertl** [comment]
Let's get this in so we can have a first beta for more testing.

**Comment by juliushaertl** [comment]
Hm, I cannot think of any. Can you test with a new user account to see if it might be related to the account settings with which editor is being used as I see NotePlain in the trace?

**Pull request merged**